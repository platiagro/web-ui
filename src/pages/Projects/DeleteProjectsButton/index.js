import React from 'react';
import PropTypes from 'prop-types';
import { Button, Popconfirm } from 'antd';
import { DeleteOutlined } from '@ant-design/icons';

const DeleteProjectsButton = ({ disabled, selectedProjects, handleClick }) => (
  <Popconfirm
    onConfirm={() => handleClick(selectedProjects)}
    title='Excluir projetos selecionados?'
    cancelText='NÃ£o'
    okText='Sim'
  >
    <Button
      disabled={disabled}
      icon={<DeleteOutlined />}
      style={{ color: '#0050B3', marginLeft: '20px' }}
    >
      Excluir selecionados
    </Button>
  </Popconfirm>
);

DeleteProjectsButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  selectedProjects: PropTypes.array.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default DeleteProjectsButton;
