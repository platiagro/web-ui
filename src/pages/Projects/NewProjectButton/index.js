import React from 'react';
import { Button } from 'antd';
import PropTypes from 'prop-types';
import { PlusOutlined } from '@ant-design/icons';

const NewProjectButton = ({ disabled, handleClick }) => (
  <Button
    className='newProjectButton'
    icon={<PlusOutlined />}
    onClick={handleClick}
    disabled={disabled}
    type='primary'
  >
    Novo Projeto
  </Button>
);

NewProjectButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default NewProjectButton;
