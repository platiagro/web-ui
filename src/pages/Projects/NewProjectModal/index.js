// CORE LIBS
import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { Form, Input, Modal } from 'antd';

/**
 * New Project Modal.
 * This component is responsible for displaying a new project modal.
 *
 * @component
 * @param {object} props Component props
 * @returns {NewProjectModal} React component
 */
const NewProjectModal = (props) => {
  // destructuring props
  const {
    visible,
    loading,
    title,
    record,
    modalValidateStatus,
    errorMessage,
    handleCloseModal,
    handleNewProject,
    handleUpdateProject,
  } = props;

  const [buttonDisabled, setButtonDisabled] = useState(false);
  const [status, setStatus] = useState(null);
  const [form] = Form.useForm();
  const inputNameRef = useRef();

  // did mount hook
  useEffect(() => {
    if (visible) {
      setTimeout(() => inputNameRef.current.select());
    } else {
      setButtonDisabled(false);
    }
    setStatus(modalValidateStatus);
  }, [modalValidateStatus, visible]);

  // FUNCTIONS
  // function to enable or disable submit button
  const onValuesChangeForm = (changedValues, allValues) => {
    if (allValues.name === '') {
      setButtonDisabled(true);
    } else {
      setButtonDisabled(false);
    }
  };
  // Function to handle form submit
  const handleSubmit = () => {
    form.validateFields().then((values) => {
      // handling create new project
      if (record !== undefined) {
        handleUpdateProject(record.uuid, values.name, values.description, true);
      } else {
        handleNewProject(values.name, values.description);
      }
    });
  };

  // FUNCTION KEYS
  const handleKeyPress = (e) => {
    if (e.key === 'Enter') {
      handleSubmit();
    } else if (e.key === 'Escape') {
      handleCloseModal();
    }
  };

  // RENDER
  return (
    // modal component
    <Modal
      visible={visible}
      title={title}
      okText={record ? 'Salvar' : 'Criar'}
      cancelText='Cancelar'
      onCancel={handleCloseModal}
      onOk={handleSubmit}
      okButtonProps={{
        disabled: buttonDisabled,
        loading,
        form: 'newProjectForm',
        key: 'submit',
        htmlType: 'submit',
      }}
      destroyOnClose
    >
      {/* form details */}
      <Form
        id='projectForm'
        layout='vertical'
        form={form}
        preserve={false}
        onValuesChange={onValuesChangeForm}
      >
        <Form.Item
          label='Qual o nome do seu projeto?'
          name='name'
          initialValue={record?.name ?? 'Novo Projeto'}
          rules={[
            {
              required: true,
              message: 'Por favor insira um nome para o projeto!',
            },
          ]}
          validateStatus={status ? modalValidateStatus : undefined}
          help={status ? errorMessage : undefined}
          autoFocus
        >
          <Input
            allowClear
            autoFocus
            ref={inputNameRef}
            onKeyUp={handleKeyPress}
            onChange={() => {
              // remove current status
              setStatus(null);
            }}
          />
        </Form.Item>
        <Form.Item
          label='Descrição (Opcional):'
          name='description'
          initialValue={record?.description}
        >
          <Input.TextArea rows={4} />
        </Form.Item>
      </Form>
    </Modal>
  );
};

// PROP TYPES
NewProjectModal.propTypes = {
  /** new project modal visible */
  visible: PropTypes.bool.isRequired,
  /** new project modal close handler */
  handleCloseModal: PropTypes.func.isRequired,
  /** new project modal new project handler */
  handleNewProject: PropTypes.func.isRequired,
  /** is loading */
  loading: PropTypes.bool.isRequired,
};

// EXPORT
export default NewProjectModal;
