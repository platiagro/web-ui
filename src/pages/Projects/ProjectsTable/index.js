import React, { useRef } from 'react';
import { SearchOutlined } from '@ant-design/icons';
import PropTypes from 'prop-types';
import {
  Button,
  Input,
  Popconfirm,
  Space,
  Tag,
  Tooltip,
  Typography,
} from 'antd';

import { CommonTable } from 'components';

const ProjectsTable = ({
  loading,
  projects,
  selectedProjects,
  handleClickDelete,
  handleClickProject,
  handleSelectProjects,
  handleFetchPaginatedProjects,
}) => {
  const searchInputRef = useRef(null);

  const handleSearch = (selectedKeys, confirm) => {
    confirm();
    handleFetchPaginatedProjects(selectedKeys[0]);
  };

  const handleReset = (clearFilters) => {
    clearFilters();
    handleFetchPaginatedProjects('');
  };

  const columnsConfig = [
    {
      title: <strong>Nome do Projeto</strong>,
      dataIndex: 'name',
      key: 'name',
      width: '20%',
      filterDropdown(filterDropdownProps) {
        const { setSelectedKeys, selectedKeys, confirm, clearFilters } =
          filterDropdownProps;

        return (
          <div style={{ padding: 8 }}>
            <Input
              ref={searchInputRef}
              placeholder={`Nome`}
              value={selectedKeys[0]}
              onPressEnter={() => handleSearch(selectedKeys, confirm)}
              style={{ width: 188, marginBottom: 8, display: 'block' }}
              onChange={(e) =>
                setSelectedKeys(e.target.value ? [e.target.value] : [])
              }
            />
            <Space>
              <Button
                type='primary'
                size='small'
                style={{ width: 90 }}
                icon={<SearchOutlined />}
                onClick={() => handleSearch(selectedKeys, confirm)}
              >
                Buscar
              </Button>
              <Button
                size='small'
                style={{ width: 90 }}
                onClick={() => handleReset(clearFilters)}
              >
                Resetar
              </Button>
            </Space>
          </div>
        );
      },
      onFilterDropdownVisibleChange: (visible) => {
        if (visible) {
          setTimeout(() => searchInputRef.current.select());
        }
      },
      filterIcon(filtered) {
        return (
          <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
        );
      },
      render(value, record) {
        return (
          <Button
            type='link'
            style={{ textAlign: 'left', width: '100%' }}
            onClick={() => handleClickProject(record.uuid)}
          >
            <Typography.Text
              style={{ color: '#0050B3', width: '100%' }}
              ellipsis
            >
              <Tooltip placement='topLeft' title={value}>
                {value}
              </Tooltip>
            </Typography.Text>
          </Button>
        );
      },
    },
    {
      title: <strong>Descrição</strong>,
      dataIndex: 'description',
      key: 'description',
      width: '30%',
      ellipsis: {
        showTitle: false,
      },
      render(value) {
        return (
          <Tooltip placement='topLeft' title={value}>
            {value}
          </Tooltip>
        );
      },
    },
    {
      title: <strong>Tags</strong>,
      dataIndex: 'tags',
      key: 'tags',
      width: '28%',
      filters: [
        {
          text: 'Experimentação',
          value: 'hasExperiment',
        },
        {
          text: 'Pré-implantação',
          value: 'hasPreDeployment',
        },
        {
          text: 'Implantado',
          value: 'hasDeployment',
        },
      ],
      onFilter(value, record) {
        /**
         * Utilizamos o value (hasDeployment, hasPreDeployment, hasExperiment)
         * para acessar a chave do objeto (hasDeployment, hasPreDeployment,
         * hasExperiment)
         */
        return record[value];
      },
      render(_, record) {
        return (
          <>
            {record.hasExperiment && <Tag color='purple'>Experimentação</Tag>}
            {record.hasPreDeployment && (
              <Tag color='volcano'>Pré-implantação</Tag>
            )}
            {record.hasDeployment && <Tag color='green'>Implantado</Tag>}
          </>
        );
      },
    },
    {
      title: <strong>Última modificação</strong>,
      dataIndex: 'updatedAt',
      key: 'updatedAt',
      width: '12%',
      sorter: (a, b) => {
        const dateA = new Date(a.updatedAt);
        const dateB = new Date(b.updatedAt);
        return dateA.getTime() - dateB.getTime();
      },
      sortDirections: ['descend', 'ascend'],
      render: (value) => new Date(value).toLocaleString(),
    },
    {
      title: <strong>Ação</strong>,
      dataIndex: 'action',
      key: 'action',
      width: '10%',
      render(_, record) {
        return (
          <Popconfirm
            title='Você tem certeza que deseja excluir esse projeto?'
            onConfirm={() => handleClickDelete(record.uuid)}
            okText='Sim'
            cancelText='Não'
          >
            <Button type='link'>Excluir</Button>
          </Popconfirm>
        );
      },
    },
  ];

  return (
    <CommonTable
      className='projectsTable'
      rowKey={(record) => record.uuid}
      rowSelection={{
        selectedRowKeys: selectedProjects,
        onChange: (selectedRowKeys) => {
          handleSelectProjects(selectedRowKeys);
        },
      }}
      dataSource={projects}
      columns={columnsConfig}
      pagination={false}
      isLoading={loading}
    />
  );
};

ProjectsTable.propTypes = {
  loading: PropTypes.bool.isRequired,
  projects: PropTypes.arrayOf(PropTypes.object).isRequired,
  selectedProjects: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleClickDelete: PropTypes.func.isRequired,
  handleClickProject: PropTypes.func.isRequired,
  handleSelectProjects: PropTypes.func.isRequired,
  handleFetchPaginatedProjects: PropTypes.func.isRequired,
};

export default ProjectsTable;
