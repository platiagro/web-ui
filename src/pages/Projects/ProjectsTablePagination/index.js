import React from 'react';
import { Pagination } from 'antd';
import PropTypes from 'prop-types';

const ProjectsTablePagination = ({
  disabled,
  currentPage,
  pageSize,
  total,
  onChange,
}) => {
  return (
    <Pagination
      disabled={disabled}
      defaultCurrent={1}
      defaultPageSize={10}
      current={currentPage}
      pageSize={pageSize}
      total={total}
      onChange={onChange}
      onShowSizeChange={onChange}
      style={{ textAlign: 'right' }}
      showSizeChanger
      pageSizeOptions={['10', '20', '30', '40', '50']}
    />
  );
};

ProjectsTablePagination.propTypes = {
  disabled: PropTypes.bool.isRequired,
  currentPage: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ProjectsTablePagination;
