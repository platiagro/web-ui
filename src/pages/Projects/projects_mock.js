// remove after tests
const projects = [
  {
    key: '1',
    name: 'workshp_ForAgri_0',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '2',
    name: 'workshp_ForAgri_1',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '3',
    name: 'workshp_ForAgri_2',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
      {
        title: 'Experimento 3',
        content: 'Content of Tab Pane 1',
        key: 'exp3',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '4',
    name: 'workshp_ForAgri_3',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
      {
        title: 'Experimento 3',
        content: 'Content of Tab Pane 1',
        key: 'exp3',
      },
      {
        title: 'Experimento 4',
        content: 'Content of Tab Pane 1',
        key: 'exp4',
      },
      {
        title: 'Experimento 5',
        content: 'Content of Tab Pane 1',
        key: 'exp5',
      },
      {
        title: 'Experimento 6',
        content: 'Content of Tab Pane 1',
        key: 'exp6',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '5',
    name: 'workshp_ForAgri_4',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
      {
        title: 'Experimento 3',
        content: 'Content of Tab Pane 1',
        key: 'exp3',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '6',
    name: 'workshp_ForAgri_5',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '7',
    name: 'workshp_ForAgri_6',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '8',
    name: 'workshp_ForAgri_7',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '9',
    name: 'workshp_ForAgri_8',
    experimentsList: [
      {
        title: 'Experimento 1',
        content: 'Content of Tab Pane 1',
        key: 'exp1',
      },
      {
        title: 'Experimento 2',
        content: 'Content of Tab Pane 1',
        key: 'exp2',
      },
    ],
    createdAt: '11/10/2019 12:59:21',
  },
  {
    key: '10',
    name: 'workshp_ForAgri_9',
    experimentsList: [],
    createdAt: '11/10/2019 12:59:21',
  },
];

export default projects;
