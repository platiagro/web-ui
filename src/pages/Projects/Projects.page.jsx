import React from 'react';

import ContentHeaderContainer from 'components/ContentHeader/ContentHeaderContainer';

import ProjectsTable from './ProjectsTable/ProjectsTableContainer';
import NewProjectModal from './NewProjectModal/NewProjectModalContainer';
import NewProjectButtonContainer from './NewProjectButton/NewProjectButtonContainer';
import DeleteProjectsButtonContainer from './DeleteProjectsButton/DeleteProjectsButtonContainer';

const ProjectsContent = () => {
  return (
    <>
      <ContentHeaderContainer
        title='Meus projetos'
        editable={false}
        backIcon={false}
      />

      <div className='contentPage'>
        <NewProjectButtonContainer />
        <DeleteProjectsButtonContainer />
        <NewProjectModal />
        <ProjectsTable />
      </div>
    </>
  );
};

export default ProjectsContent;
