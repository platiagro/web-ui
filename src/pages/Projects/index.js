import React from 'react';

import './style.scss';

import { Divider, Button, Empty } from 'antd';

import NewProjectModal from '../../components/NewProjectModal';
import ProjectsTable from '../../components/ProjectsTable';

import meh from '../../assets/meh.svg';

class Projects extends React.Component {
  constructor(props) {
    super(props);

    // remove after tests
    const projects = [
      {
        key: '1',
        projectName: 'workshp_ForAgri_0',
        experimentsList: [
          'Experimento 0',
          'Experimento 1',
          'Experimento 2',
          'Experimento 3',
          'Experimento 4',
          'Experimento 5',
          'Experimento 6',
        ],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '2',
        projectName: 'workshp_ForAgri_1',
        experimentsList: ['Experimento 0', 'Experimento 1', 'Experimento 2'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '3',
        projectName: 'workshp_ForAgri_2',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '4',
        projectName: 'workshp_ForAgri_3',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '5',
        projectName: 'workshp_ForAgri_4',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '6',
        projectName: 'workshp_ForAgri_5',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '7',
        projectName: 'workshp_ForAgri_6',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '8',
        projectName: 'workshp_ForAgri_7',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '9',
        projectName: 'workshp_ForAgri_8',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
      {
        key: '10',
        projectName: 'workshp_ForAgri_9',
        experimentsList: ['Experimento 0'],
        created: '11/10/2019 12:59:21',
      },
    ];

    projects.forEach((project) => {
      const projectAux = project;
      projectAux.experiments = project.experimentsList.join(', ');
    });

    this.state = {
      projectList: projects,
      selectedRowKeys: [],
      modalIsVisible: false,
    };

    this.renderBody = this.renderBody.bind(this);
    this.showModal = this.showModal.bind(this);
    this.hideModal = this.hideModal.bind(this);
  }

  handleCreate = () => {
    const { form } = this.formRef.props;
    form.validateFields((err, values) => {
      if (err) {
        return;
      }

      // eslint-disable-next-line
      console.log('Received values of form: ', values);
      form.resetFields();
      this.setState({ modalIsVisible: false });
    });
  };

  saveFormRef = (formRef) => {
    this.formRef = formRef;
  };

  onSelectChange = (selectedRowKeys) => {
    // eslint-disable-next-line
    console.log('selectedRowKeys changed: ', selectedRowKeys);
    this.setState({ selectedRowKeys });
  };

  showModal() {
    this.setState({ modalIsVisible: true });
  }

  hideModal() {
    const { form } = this.formRef.props;

    this.setState({ modalIsVisible: false });

    form.resetFields();
  }

  renderBody() {
    const { projectList, selectedRowKeys } = this.state;

    const rowSelection = {
      selectedRowKeys,
      onChange: this.onSelectChange,
    };

    return projectList.length === 0 ? (
      <Empty
        image={meh}
        imageStyle={{
          height: 60,
        }}
        description={
          <span>
            <span>
              <strong>Nenhum projeto foi criado</strong>
            </span>
            <br />
            <span>Clique no botão &quot;Novo Projeto&quot; para começar</span>
          </span>
        }
      />
    ) : (
      <ProjectsTable projectList={projectList} rowSelection={rowSelection} />
    );
  }

  render() {
    const { modalIsVisible, selectedRowKeys } = this.state;

    const hasSelected = selectedRowKeys.length > 0;

    return (
      <div className='project-page'>
        <NewProjectModal
          wrappedComponentRef={this.saveFormRef}
          visible={modalIsVisible}
          onCancel={this.hideModal}
          onCreate={this.handleCreate}
        />

        <div className='header'>
          <Button onClick={this.showModal} type='primary' icon='plus'>
            Novo Projeto
          </Button>

          <Divider type='vertical' />

          <Button disabled={!hasSelected}>Excluir</Button>
        </div>

        <Divider />

        <div className='body'>{this.renderBody()}</div>
      </div>
    );
  }
}

export default Projects;
