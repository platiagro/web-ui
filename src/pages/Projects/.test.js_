import React from 'react';

import { shallow } from 'enzyme';

import Projects from '.';

describe('Page Projects should', () => {
  it('renders without crashing', () => {
    shallow(<Projects />);
  });

  it('renders html correctly', () => {
    const projectsShallowed = shallow(<Projects />);

    expect(projectsShallowed).toMatchSnapshot();
  });
});
