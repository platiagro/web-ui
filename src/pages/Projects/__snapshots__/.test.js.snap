// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Projects should renders html correctly 1`] = `
<div
  className="project-page"
>
  <Form(NewProjectModal)
    onCancel={[Function]}
    onCreate={[Function]}
    visible={false}
    wrappedComponentRef={[Function]}
  />
  <div
    className="header"
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      icon="plus"
      loading={false}
      onClick={[Function]}
      type="primary"
    >
      Novo Projeto
    </Button>
    <Divider
      type="vertical"
    />
    <Button
      block={false}
      disabled={true}
      ghost={false}
      htmlType="button"
      loading={false}
    >
      Excluir
    </Button>
  </div>
  <Divider />
  <div
    className="body"
  >
    <ProjectsTable
      projectList={
        Array [
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0, Experimento 1, Experimento 2, Experimento 3, Experimento 4, Experimento 5, Experimento 6",
            "experimentsList": Array [
              "Experimento 0",
              "Experimento 1",
              "Experimento 2",
              "Experimento 3",
              "Experimento 4",
              "Experimento 5",
              "Experimento 6",
            ],
            "key": "1",
            "projectName": "workshp_ForAgri_0",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0, Experimento 1, Experimento 2",
            "experimentsList": Array [
              "Experimento 0",
              "Experimento 1",
              "Experimento 2",
            ],
            "key": "2",
            "projectName": "workshp_ForAgri_1",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "3",
            "projectName": "workshp_ForAgri_2",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "4",
            "projectName": "workshp_ForAgri_3",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "5",
            "projectName": "workshp_ForAgri_4",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "6",
            "projectName": "workshp_ForAgri_5",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "7",
            "projectName": "workshp_ForAgri_6",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "8",
            "projectName": "workshp_ForAgri_7",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "9",
            "projectName": "workshp_ForAgri_8",
          },
          Object {
            "created": "11/10/2019 12:59:21",
            "experiments": "Experimento 0",
            "experimentsList": Array [
              "Experimento 0",
            ],
            "key": "10",
            "projectName": "workshp_ForAgri_9",
          },
        ]
      }
      rowSelection={
        Object {
          "onChange": [Function],
          "selectedRowKeys": Array [],
        }
      }
    />
  </div>
</div>
`;
